File: c/prob31.c
#include <stdio.h>

int main() {
    int c100, c50, c20, c10, c5, c2, c1;
    int total = 1; // The 200p coin is one possiblity
    for (c100=0; c100<3; c100++) {
        for (c50=0; c50<5; c50++) {
            for (c20=0; c20<11; c20++) {
                for (c10=0; c10<21; c10++) {
                    for (c5=0; c5<41; c5++) {
                        for (c2=0; c2<101; c2++) {
                            for (c1=0; c1<201; c1++) {
                                if (((c100 * 100) + (c50 * 50) + (c20 * 20) +
                                     (c10 * 10) + (c5 * 5) + (c2 * 2) + c1)
                                    == 200) {
                                    total += 1;
                                }
                            }
                        }
                    }
                }                    
            }
        }
    }
    printf("%d\n", total);
    return 0;
}

File: clojure/problems/prob31.clj
(ns prob31)

(defn prob31 [amt]
  (let [end (inc amt)]
    (reduce + 
            (for [c200 (range 0 end 200)
                  c100 (range 0 (- end c200) 100)
                  c50 (range 0 (- end c200 c100) 50)
                  c20 (range 0 (- end c200 c100 c50) 20)
                  c10 (range 0 (- end c200 c100 c50 c20) 10)
                  c5 (range 0 (- end c200 c100 c50 c20 c10) 5)
                  c2 (range 0 (- end c200 c100 c50 c20 c10 c5) 2)
                  c1 (range 0 (- end c200 c100 c50 c20 c10 c5 c2) 1)
                  :when (= amt (+ c200 c100 c50 c20 c10 c5 c2 c1))]
              1))))
